tests/validation/cattlevalidationtest/core/test_machine_amazon.py

```diff
@@ -0,0 +1,531 @@
+import boto.ec2
+import logging
+import traceback
+
+from common_fixtures import *  # NOQA
+
+DEFAULT_TIMEOUT = 450
+
+access_key = os.environ.get('AMAZON_ACCESSKEY')
+secret_key = os.environ.get('AMAZON_SECRETKEY')
+
+# Use amazon settings from environment variables , if set
+vpc_id = os.environ.get('AMAZON_VPC_ID', "vpc-eb2d928e")
+subnet_id = os.environ.get('AMAZON_SUBNET_ID', "subnet-5775c020")
+security_group = os.environ.get('AMAZON_SECURITY_GROUP', "test")
+zone = os.environ.get('AMAZON_ZONE', "a")
+region = os.environ.get('AMAZON_REGION', "us-west-2")
+ami = os.environ.get('AMAZON_AMI', "ami-5189a661")
+vpc_id_in_east = os.environ.get('AMAZON_VPC_ID', "vpc-44144621")
+subnet_id_in_east = os.environ.get('AMAZON_SUBNET_ID', "subnet-22d1bd55")
+
+size = "32"
+instance_type = "t2.small"
+
+
+# Default values set by Docker Machine
+default_image_in_west = "ami-898dd9b9"
+default_image_in_east = "ami-4ae27e22"
+default_security_group = "docker-machine"
+default_instance_type = "t2.micro"
+default_root_size = "16"
+default_zone = "a"
+default_region = "us-east-1"
+
+# Amazon error message validation
+
+error_msg_auth_failure = "code=401"
```

Comment:
These are error messages that we get from docker machine. There is a possibility that with a later version of docker machine , these error messages may change. To minimize false failures due to change in error message across different versions of docker machine , the check for the entire message is not done instead I check for  a part of it.